server/src/diskstore/snapshot.rs

```diff
@@ -94,35 +94,34 @@ impl<'a> SnapshotEngine<'a> {
                         let entry = entry?;
                         let path = entry.path();
                         // We'll skip the directory that contains remotely created snapshots
-                        if path.is_dir()
-                            && (path.parent().expect("Couldn't get parent path!")
-                                != Path::new("remote"))
-                        {
+                        if path.is_dir() && path != PathBuf::from("data/snapshots/remote") {
                             // If the entry is not a directory then some other
                             // file(s) is present in the directory
                             return Err(
                                 "The snapshot directory contains unrecognized files/directories"
                                     .into(),
                             );
                         }
-                        let fname = entry.file_name();
-                        let file_name = if let Some(good_file_name) = fname.to_str() {
-                            good_file_name
-                        } else {
-                            // The filename contains invalid characters
-                            return Err(
+                        if !path.is_dir() {
+                            let fname = entry.file_name();
+                            let file_name = if let Some(good_file_name) = fname.to_str() {
+                                good_file_name
+                            } else {
+                                // The filename contains invalid characters
+                                return Err(
```

Comment:
This change is also significant and fixes the remote snapshot directory item parsing logic